packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -427,6 +432,70 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.updateCustomer(uid, email);
   }
 
+  /**
+   * Validate plan metadata on update
+   */
+  async handlePlanUpdate(request: AuthRequest, event: Stripe.Event) {
+    const plan = event.data.object as Stripe.Plan;
+    const products = await this.stripeHelper.allProducts();
+    let product = products.find((p) => p.product_id === plan.product);
+    if (!product) {
+      this.log.error(
+        `handlePlanUpdate - Could not locate Product ${plan.product}`,
+        { item: plan }
+      );
+    } else {
+      const result = subscriptionProductMetadataValidator.validate({
+        ...product.product_metadata,
+        ...plan.metadata,
+      });
+
+      if (result?.error) {
+        const msg = `handlePlanUpdate - Plan "${plan.id}"'s metadata failed validation; After Plan update.`;
```

Comment:
nit: 'After Plan update' seems a little redundant since the message also has 'handlePlanUpdate'.